common/dynamicconfig/cachedvalue.go

```diff
@@ -0,0 +1,59 @@
+package dynamicconfig
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"go.temporal.io/server/common/log/tag"
+)
+
+// GlobalCachedTypedValue holds a cached value of a GlobalTypedSetting after applying a supplied convert function.
+// This type should be deleted after caching is added to GlobalTypedSettingWithConverter.
```

Comment:
let's not presume the final API yet...
```suggestion
// This type should be deleted after dynamicconfig handles conversion more efficiently.
```